main.tf

```diff
@@ -0,0 +1,73 @@
+module "label" {
+  source      = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.5.3"
+  namespace   = "${var.namespace}"
+  name        = "${var.name}"
+  stage       = "${var.stage}"
+  environment = "${var.environment}"
+  delimiter   = "${var.delimiter}"
+  attributes  = "${var.attributes}"
+  tags        = "${var.tags}"
+
+  additional_tag_map = {
+    "propagate_at_launch" = "true"
+  }
+}
+
+resource "aws_launch_configuration" "default" {
+  count = "${var.launch_configuration_enabled == "true" ? 1: 0}"
+
+  name_prefix                 = "${format("%s%s", module.label.id, var.delimiter)}"
+  image_id                    = "${var.image_id}"
+  instance_type               = "${var.instance_type}"
+  iam_instance_profile        = "${var.iam_instance_profile}"
+  key_name                    = "${var.key_name}"
+  security_groups             = ["${var.security_groups}"]
+  associate_public_ip_address = "${var.associate_public_ip_address}"
+  user_data_base64            = "${var.user_data_base64}"
+  enable_monitoring           = "${var.enable_monitoring}"
+  ebs_optimized               = "${var.ebs_optimized}"
+  root_block_device           = "${var.root_block_device}"
```

Comment:
It might be more beneficial for an end user to provide values for storage type, disk size, and boolean for delete on termination as variables for `root_block_device`. Looking at the syntax in the example with the way it's currently written I think these three variables would be a better experience as a user of the module.

Reply:
This is hard to explain without providing an example, please let me know if you need clarification.